mssql-plugin/src/main/java/io/cdap/plugin/mssql/SqlFieldsValidator.java

```diff
@@ -35,17 +35,30 @@ public boolean isFieldCompatible(Schema.Field field, ResultSetMetaData metadata,
     Schema.LogicalType fieldLogicalType = fieldSchema.getLogicalType();
 
     int sqlType = metadata.getColumnType(index);
+    String sqlTypeName = metadata.getColumnTypeName(index);
     boolean isSigned = metadata.isSigned(index);
     int precision = metadata.getPrecision(index);
 
     // Handles datetime datatypes
     // Case when Timestamp maps to datetime
     // Case when Datetime2 maps to datetime
-    // Case when DatetimeOffset maps to datetime
     if ((sqlType == Types.TIMESTAMP || sqlType == SqlServerSourceSchemaReader.DATETIME_OFFSET_TYPE)
-            && fieldLogicalType.equals(Schema.LogicalType.DATETIME)) {
+      && fieldLogicalType.equals(Schema.LogicalType.DATETIME)) {
       return true;
     }
+
+    // Case when datetimeoffset maps to timestamp
+    if (sqlType == SqlServerSourceSchemaReader.DATETIME_OFFSET_TYPE &&
+      fieldLogicalType.equals(Schema.LogicalType.TIMESTAMP_MICROS)) {
+      return true;
+    }
+
+    // Case when smalldatetime maps to datetime
+    if (sqlTypeName.equalsIgnoreCase("smalldatetime")
+      && fieldLogicalType.equals(Schema.LogicalType.DATETIME)) {
+      return true;
+    }
```

Comment:
Why is this required, If I understand correctly the above block at line 45, should already cover for this use case.

Reply:
removed this block